### DISCUSSION TITLE
Brightness works but no volume control (BenQ PD2725U)

### DISCUSSION THREAD

**Discussion by Mactastic1-5** [MonitorControl/MonitorControl#576] [Q&A] [answered]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] I have read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Volume controls aren't working on BenQ PD2725U.

### Expected behavior

Volume in OSD is supposed to change to the value specified by MonitorControl's Menu item and/or keyboard.

### Anything else?

![Screen Shot 2021-09-10 at 3 41 02 PM](https://user-images.githubusercontent.com/17728506/132908531-1b0e9d72-5f39-4936-9796-c724d472d669.png)

When I use m1ddc, it doesn't change the volume. Output is below.
```
./m1ddc display list
1 - BenQ PD2725U (omitted)
./m1ddc display 1 chg volume 40
100
```

### Environment Information (please complete the following information)

```markdown
- macOS version: 11.5.2 (20G95)
- Mac model: Mac mini
- MonitorControl version: 3.0.0
- Monitor(s): BenQ PD2725U
- Monitor Cable(s)/Connection(s): Thunderbolt (USB Type-C)
- Apple Silicon/M1 (yes or no): Yes
```

**Comment by waydabber** [comment]
Hey @MacarioPatrick, if m1ddc does not work either, then it is probably a display issue and cannot be solved. An other app you might want to try is [Lunar](https://lunar.fyi) by @alin23 as it has a free mode and see if that can control volume. If that doesn't work either, then probably there is no much hope I am afraid.

Please report back so we could close the issue or proceed with a fix if you find Lunar working.

**Comment by Mactastic1-5** [comment]
> Hey @MacarioPatrick, if m1ddc does not work either, then it is probably a display issue and cannot be solved. An other app you might want to try is [Lunar](https://lunar.fyi) by @alin23 as it has a free mode and see if that can control volume. If that doesn't work either, then probably there is no much hope I am afraid.
> 
> Please report back so we could close the issue or proceed with a fix if you find Lunar working.

I've tried Lunar, and it doesn't control volume.
![Screen Shot 2021-09-10 at 5 38 33 PM](https://user-images.githubusercontent.com/17728506/132920481-49d51bd0-b3e6-42c6-8530-29a3b9c1d9c7.png)

I have the Satechi Stand & Hub for Mac mini with SSD, but it's plugged directly into a Thunderbolt (USB Type-C) port on the Mac mini. Could this be interfering or something?

I'd also like to add that brightness controls reflect the brightness value in OSD, but for some reason, not volume.

**Comment by waydabber** [comment]
No, I don't think it interferes. If brightness control works but volume doesn't then there is nothing wrong with DDC communication, only the display doesn't respond to volume commands. It is probably a display firmware issue. :( It might be possible that the display would respond to some different values or needs a speaker select command. Figuring this out would need quite bit of tinkering with the display and still might not result in a solution.

Lunar has a nice CLI (command line) mode which allows passing custom hex value control commands to the display. You could play with that. You can find out the possible commands from the [VESA specification](https://github.com/reitermarkus/DDC.swift/blob/master/doc/MCCS-2.2a.pdf).

**Comment by waydabber** [comment]
(I'll convert this issue to a discussion since it is not an issue that can be fixed but might be beneficial for others)

**Reply by Mactastic1-5** [reply]
This is what BenQ technical support said:

"The VCP code of volume adjustment is not defined on PD2725U, so you can't adjust the volume via DDC/CI (Display Pilot, Lunar, volume control, or other similar applications.) You can use other applications to control those functions that were defined with VCP codes (such as Brightness) for Display Pilot."

**Reply by alin23** [reply]
It's the first time I see such a technical answer from a vendor 😅

Well at least now we know, they just didn't bother to implement the audio control code inside their firmware. There are even more BenQ (and some Lenovo) monitors that have this problem from my experience with Lunar users.

**Reply by Mactastic1-5** [reply]
> It's the first time I see such a technical answer from a vendor 😅
> 
> 
> 
> Well at least now we know, they just didn't bother to implement the audio control code inside their firmware. There are even more BenQ (and some Lenovo) monitors that have this problem from my experience with Lunar users.

They asked their research and development department for an answer and they relayed it back to me.

**Comment by Mactastic1-5** [comment, marked as answer]
@waydabber I use the Super Volume Keys feature in an application called SoundSource to control the volume of BenQ PD2725U. Maybe you can implement something similar into MonitorControl.

**Reply by waydabber** [reply]
Yes, SoundSource uses a virtual audio device as a filter to control the intensity of the audio bitstream thus controlling the volume. Unfortunately this is very much out of scope for MonitorControl, it is easier to recommend users to use existing solutions. Those who don't want to purchase SoundSource, [BlackHole](https://github.com/ExistentialAudio/BlackHole) might be a good (albeit not as refined) alternative.

**Comment by beamercola** [comment]
@MacarioPatrick did you ever find relief from this?  I just bought this display and kinda shocked i can't change the volume without digging in the menu

**Reply by Mactastic1-5** [reply]
> @MacarioPatrick did you ever find relief from this?  I just bought this display and kinda shocked i can't change the volume without digging in the menu

You have to use SoundSource or similar. It isn't free.
